priam/src/main/java/com/netflix/priam/restore/AbstractRestore.java

```diff
@@ -95,161 +89,94 @@ public AbstractRestore(
-                logger.info("No valid snapshot meta file found, Restore Failed.");
-                instanceState.getRestoreStatus().setExecutionEndTime(LocalDateTime.now());
-                instanceState.setRestoreStatus(Status.FAILED);
+                instanceState.endRestore(Status.FAILED, LocalDateTime.now());
                 return;
             }
-
-            logger.info(
-                    "Snapshot Meta file for restore {}", latestValidMetaFile.get().getRemotePath());
-            instanceState
-                    .getRestoreStatus()
-                    .setSnapshotMetaFile(latestValidMetaFile.get().getRemotePath());
-
+            logger.info("Meta file for restore {}", latestValidMetaFile.get().getRemotePath());
+            instanceState.setRestoreMetaFile(latestValidMetaFile.get().getRemotePath());
             List<AbstractBackupPath> allFiles =
                     BackupRestoreUtil.getMostRecentSnapshotPaths(
                             latestValidMetaFile.get(), metaProxy, pathProvider);
             if (!config.skipIncrementalRestore()) {
-                allFiles.addAll(
-                        BackupRestoreUtil.getIncrementalPaths(
-                                latestValidMetaFile.get(), dateRange, metaProxy));
+                DateUtil.DateRange incrementalDateRange =
+                        new DateUtil.DateRange(
+                                latestValidMetaFile.get().getLastModified(),
+                                dateRange.getEndTime());
+                allFiles.addAll(metaProxy.getIncrementals(incrementalDateRange));
             }
-
-            // Download snapshot which is listed in the meta file.
-            List<Future<Path>> futureList = new ArrayList<>();
-            futureList.addAll(download(allFiles.iterator(), false));
-
-            // Wait for all the futures to finish.
+            List<Future<Path>> futureList = new ArrayList<>(download(allFiles.iterator()));
             waitForCompletion(futureList);
-
-            // Given that files are restored now, kick off post restore hook
-            logger.info("Starting post restore hook");
-            postRestoreHook.execute();
-            logger.info("Completed executing post restore hook");
-
-            // Declare restore as finished.
-            instanceState.getRestoreStatus().setExecutionEndTime(LocalDateTime.now());
-            instanceState.setRestoreStatus(Status.FINISHED);
-
-            // Start cassandra if restore is successful.
-            if (!config.doesCassandraStartManually()) cassProcess.start(true);
-            else
-                logger.info(
-                        "config.doesCassandraStartManually() is set to True, hence Cassandra needs to be started manually ...");
+            List<String> failedImports = cassOps.importAll(config.getRestoreDataLocation());
+            if (!failedImports.isEmpty()) {
+                instanceState.endRestore(Status.FAILED, LocalDateTime.now());

```

Comment:
do we want to use the list of failed imports somewhere to log what failed and be able to resume? instead of starting over from the beginning? 

Also if we only need the boolean why return the list? I would rather us use the list but if we are not going to I mean.